### PR TITLE
feat!: ability to extend the Tailwind config

### PR DESCRIPTION
**Pull request by Atinux**
This will be a breaking change since it requires the `tailwind.config.js` to let Nuxt modules extend the tailwind config.

I am also depending on the next version of defu that handle array merging: https://github.com/nuxt-contrib/defu/pull/18, you need to `yarn link defu` in order to make it work.

This has two benefits:
- `@nuxtjs/tailwindcss` now configures the `purge` option of TailwindCSS and let the user extend it easily while keeping a minimal `tailwind.config.js` template
- Modules / Themes can easily extend the user tailwind config (adding Typography plugin, settings colors, etc)
- Updating the `tailwind.config.js` will actually do a full Nuxt server restart but will actually fix the plugins update issue
- Breaking change: `build.postcss.plugins` as `Array` is not supported anymore

TODOS:
- [x] Update README
- [x] Update tests

### CODE REVIEW

File: lib/module.js [hunk 1]
```diff
@@ -1,51 +1,65 @@
...
     ** Set PostCSS config
     */
-    const { postcss } = this.options.build
+    const { postcss } = nuxt.options.build
 
     postcss.preset.stage = 1 // see https://tailwindcss.com/docs/using-with-preprocessors#future-css-features
     postcss.plugins = postcss.plugins || {}
 
+    // Get and extend the Tailwind config
+    let tailwindConfig = require(configPath)
```
Comment (pi0) [inline lib/module.js:51]: What would be the advantage of require external file over reading from nuxt.options.tailwind? This makes configuration of module inconsistent with others

Comment (pi0) [reply lib/module.js:51]: or maybe we can support both configuration sources?

Comment (Atinux) [reply lib/module.js:51]: Well the module options is different than the Tailwind config (and keeping the standard of using `tailwind.config.js`)

**Comment by Atinux** [comment]
Ready to review @pi0 :)

**Comment by Atinux** [comment]
Docs preview: https://deploy-preview-133--nuxt-tailwindcss.netlify.app/

File: docs/static/sw.js [hunk 2]
```diff
@@ -0,0 +1 @@
+// THIS FILE SHOULD NOT BE VERSION CONTROLLED
```
Comment (pi0) [inline docs/static/sw.js:1]: this can be gitignored

File: docs/nuxt.config.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import theme from '@nuxt/content-theme-docs'
+
+export default theme({
+  generate: {
+    routes: ['/'],
+    fallback: '404.html'
```
Comment (pi0) [inline docs/nuxt.config.js:6]: not related to this PR but I see fallback is set for all docs. Maybe better defaulting for theme preset? (which is still overridable)

Comment (Atinux) [reply docs/nuxt.config.js:6]: Full agree on this, cc @benjamincanac

File: docs/static/sw.js (hunk 2 above)
Comment (Atinux) [reply docs/static/sw.js]: Done, nice catch bro ❤️

File: docs/nuxt.config.js (hunk 3 above)
Comment (benjamincanac) [reply docs/nuxt.config.js:6]: I agree too, is this just for Netlify or will it work for Vercel too?

Comment (Atinux) [reply docs/nuxt.config.js:6]: Not the fallback since it is configured to `200.html` by default.

**Comment by web2033** [comment]
Wow, you're moving too fast)
I thought it was a bug when `tailwind.config.js` wasn't created automatically...

**Comment by jessecogollo** [comment]
👏 great job.

**Comment by evolve2k** [comment]
Hi seeking a little clarity.

If we're updating to this and have an existing tailwind.config.js file, what do I need to update to account for the breaking changes mentioned here? Very clear that lots of work has gone in, not so clear exactly what I need to adjust for to addressing breaking changes.

Thx

**Pull request merged**